src/lib/optimizer/optimizer.cpp

```diff
@@ -89,6 +90,7 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   final_batch.add_rule(std::make_shared<ChunkPruningRule>());
   final_batch.add_rule(std::make_shared<ConstantCalculationRule>());
   final_batch.add_rule(std::make_shared<IndexScanRule>());
+  final_batch.add_rule(std::make_shared<ColumnPruningRule>());
```

Comment:
Shouldn't this be done quite early so that following rules run on a simpler LQP?

Reply:
Actually, this rule will make the LQP more complex, since it adds nodes.